corehq/ex-submodules/pillowtop/utils.py

```diff
@@ -268,11 +268,14 @@ def _convert_rev_to_int(rev):
 
 def ensure_document_exists(change):
     """
-    Ensures that the document recorded in Kafka exists and is properly returned
+    This is to ensure that the Couch document exists in the Couch database when the
+    change is processed. We only care about the scenario where the document is missing.
 
     :raises: DocumentNotFoundError - Raised when the document is not found
     """
-    doc = change.get_document()
-    if doc is None:
-        pillow_logging.warning("Unable to get document from change: {}".format(change))
-        raise DocumentNotFoundError()  # force a retry
+    try:
+        change.get_document(raise_error=True)
+    except DocumentMissingError:
+        raise
+    except DocumentNotFoundError:
+        pass
```

Comment:
If we don't raise the error here won't it error somewhere else because the doc returned is `None`